### PR TITLE
sonic 1.2.4 (new formula)

### PR DESCRIPTION
**Pull request by ankane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Fast, lightweight search server

### CODE REVIEW

File: Formula/sonic.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Sonic < Formula
+  desc "Fast, lightweight & schema-less search backend"
+  homepage "https://github.com/valeriansaliou/sonic"
+  url "https://github.com/valeriansaliou/sonic/archive/v1.2.4.tar.gz"
+  sha256 "290da969d39260abe5772fc7f084d256a8c0a53c702fe9eb44593d43a179b9d1"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "."
```
Comment (SMillerDev) [inline Formula/sonic.rb:10]: Please use `std_cargo_args`

File: Formula/sonic.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  test do
+    port = free_port
+
+    cp etc/"sonic.cfg", testpath/"config.cfg"
+    inreplace "config.cfg", ":1491", ":#{port}"
+    inreplace "config.cfg", "#{var}/sonic", "."
+
+    fork { exec bin/"sonic" }
+    sleep 2
+    system "nc", "-z", "localhost", port
```
Comment (SMillerDev) [inline Formula/sonic.rb:58]: Is there no query you can run against the server?

**Review by SMillerDev — DISMISSED** [review]

File: Formula/sonic.rb (hunk 1 above)
Comment (ankane) [reply Formula/sonic.rb]: Done

File: Formula/sonic.rb (hunk 2 above)
Comment (ankane) [reply Formula/sonic.rb:58]: Sorry, meant to mention that in notes. It uses its own wire protocol.

Comment (ankane) [reply Formula/sonic.rb:58]: Ref: https://github.com/valeriansaliou/sonic#perform-searches-and-manage-objects

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**